### PR TITLE
[SPARK-46358][CONNECT] Simplify the condition check in the `ResponseValidator#verifyResponse` 

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This PR has made the following refactoring to the `verifyResponse` function in `ResponseValidator`:
1. The check condition `response.hasField(field)` is moved before getting `value`, and only when `response.hasField(field)` is true, `value` is obtained, which seems more in line with the existing comments.
2. Removed the `value != ""` condition check in the case match, because only when `value.nonEmpty` is true will it enter the `if` branch, and the condition `value.nonEmpty` has already covered the check for `value != ""`.
3. The condition check `value != id` is moved inside `case Some(id)`. After the modification, an `IllegalStateException` will still be thrown when the id exists and `value != id`, but `serverSideSessionId` will no longer be reassigned when the id exists and `value == id`.
4. Removed the redundant `toString` operation when reassigning `serverSideSessionId`, because `value` is String type.
5. Removed the No-op `case _` match, because it is unreachable code after the above modifications.

### Why are the changes needed?
Simplify the condition check in the `verifyResponse` function

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/ResponseValidator.scala [hunk 1]
```diff
@@ -35,18 +35,20 @@ class ResponseValidator extends Logging {
     val field = response.getDescriptorForType.findFieldByName("server_side_session_id")
     // If the field does not exist, we ignore it. New / Old message might not contain it and this
     // behavior allows us to be compatible.
-    if (field != null) {
+    if (field != null && response.hasField(field)) {
```
Comment (LuciferYang) [inline connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/ResponseValidator.scala:38]: cc @grundprinzip 
also cc @dongjoon-hyun

**Review by dongjoon-hyun — APPROVED** [review]
+1, all changes look reasonable to me. Thanks, @LuciferYang .

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun ~

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun ~

**Review by beliefer — COMMENTED** [review]
Late LGTM.

**Comment by LuciferYang** [comment]
thanks @beliefer

**Pull request closed without merging**